crates/sui-framework/packages/sui-framework/sources/event.move

```diff
@@ -37,6 +40,70 @@ module sui::event;
 /// phantom parameters, eg `emit(MyEvent<phantom T>)`.
 public native fun emit<T: copy + drop>(event: T);
 
+public use fun destroy_stream as EventStream.destroy;
+public use fun destroy_cap as EventStreamCap.destroy;
+public use fun emit_authenticated as EventStreamCap.emit;
+
+#[allow(unused_field)]
+public struct EventStreamHead has store {
+    /// Merkle root for all events in the current checkpoint.
+    root: vector<u8>,
+    /// Hash of the previous version of the head object.
+    prev: vector<u8>,
+}
+
+// A unique identifier for an event stream.
+public struct EventStream has key, store {
+    id: UID,
+}
+
+public fun new_event_stream(ctx: &mut TxContext): EventStream {
+    EventStream {
+        id: object::new(ctx),
+    }
+}
+
+public fun destroy_stream(stream: EventStream) {
+    let EventStream { id } = stream;
+    id.delete();
+}
+
+/// A capability to write to a given event stream
+public struct EventStreamCap has key, store {
+    id: UID,
+    stream_id: address,
+}
+
+public fun get_cap(stream: &EventStream, ctx: &mut TxContext): EventStreamCap {
```

Comment:
The impression I get here is that `EventStream`s should generally be guarded (wrapped) by the packages that create them, because otherwise, anyone could create these capabilities -- is that right?

I would defer to people who write Move more regularly but this seems slightly unconventional, compared to an API where you construct a stream and its capability together, and then potentially allow the capability holder to create copies of that capability, etc.

Reply:
Yes - if you leak an `EventStream` or an `EventStreamCap` you have leaked the ability to add new events to that stream.

I am happy to defer to those who have strong opinions about idiomatic Move